h2o-r/h2o-package/R/infogram.R

```diff
@@ -0,0 +1,526 @@
+  if (!missing(top_n_features))
+    parms$top_n_features <- top_n_features
+  if (!missing(compute_p_values))
+    parms$compute_p_values <- compute_p_values
+
+  if (!missing(infogram_algorithm_params))
+      parms$infogram_algorithm_params <- as.character(toJSON(infogram_algorithm_params, pretty = TRUE))
+  h2o.show_progress() # enable progress bar explicitly
+
+  # Build segment-models specific parameters
+  segment_parms <- list()
+  if (!missing(segment_columns))
+    segment_parms$segment_columns <- segment_columns
+  if (!missing(segment_models_id))
+    segment_parms$segment_models_id <- segment_models_id
+  segment_parms$parallelism <- parallelism
+
+  # Error check and build segment models
+  segment_models <- .h2o.segmentModelsJob('infogram', segment_parms, parms, h2oRestApiVersion=3)
+  return(segment_models)
+}
+
+
+#' Extract the admissible attributes/predictors out of the H2O Infogram Model.
+#'
+#' @param model A fitted \linkS4class{H2OInfogram} object.
+#' @export 
+h2o.get_admissible_attributes <- function(model) {
+  if (is(model, "H2OInfogram") && (model@algorithm == "infogram"))
+    return(model@admissible_features)
+}
+
+#' Plot an H2O Infogram
+#'
+#' Plots the Infogram for an H2OInfogram object.
+#'
+#' @param x A fitted \linkS4class{H2OInfogram} object.
+#' @param ... additional arguments to pass on.
+#' @return A ggplot2 object.
+#' @seealso \code{\link{h2o.infogram}}
+#' @examples
+#' \dontrun{
+#' h2o.init()
+#' 
+#' # Convert iris dataset to an H2OFrame
+#' train <- as.h2o(iris)
+#' 
+#' # Create and plot infogram
+#' ig <- h2o.infogram(y = "Species", training_frame = train)
+#' plot(ig)
+#' 
+#' }
+#' @export
+plot.H2OInfogram <- function(x, ...) {
+  .check_for_ggplot2() # from explain.R
+  varargs <- list(...)
+  if ("title" %in% names(varargs)) {
+    title <- varargs$title
+  } else {
+    title <- "Infogram"
+  }
+  if ("total_information" %in% names(x@admissible_score)) {
+    # core infogram
+    xlab <- "Total Information"
+    ylab <- "Net Information"
+    xthresh <- x@total_information_threshold
+    ythresh <- x@net_information_threshold
+  } else {
+    # fair infogram
+    xlab <- "Relevance Index"
+    ylab <- "Safety Index"
+    xthresh <- x@relevance_index_threshold
+    ythresh <- x@safety_index_threshold
+  }
+  df <- as.data.frame(x@admissible_score)
+  # use generic names for x, y for easier ggplot code
+  names(df) <- c("column",
+                 "admissible",
+                 "admissible_index",
+                 "ig_x",
+                 "ig_y",
+                 "raw")
+  ggplot2::ggplot(data = df, ggplot2::aes_(~ig_x, ~ig_y)) +
+    ggplot2::geom_point() +
+    ggplot2::geom_polygon(ggplot2::aes(x, y), data = data.frame(
+      x = c(xthresh, xthresh, -Inf, -Inf, Inf, Inf, xthresh),
+      y = c(ythresh, Inf, Inf, -Inf, -Inf, ythresh, ythresh)
+    ), alpha = 0.1, fill = "#CC663E") +
+    ggplot2::geom_path(ggplot2::aes(x, y), data = data.frame(
```

Comment:
same here

Reply:
@tomasfryda Can we re-use the same `.data` object here, which was already set to `NULL` and then used in the polygon above?  Or do I need to declare a second `.data2 = NULL`?

Reply:
The `.data` is meant to be reused. The `.data <- NULL` is there just so that CRAN doesn't complain about uninitialized variable.

Reply:
Pushed a fix.